modules/integration/tests-integration/tests-security-manager/src/test/java/org/wso2/carbon/appserver/integration/test/server/security/manager/Carbon15560TestCase.java

```diff
@@ -0,0 +1,161 @@
+        return new Object[][]{
+                new Object[]{TestUserMode.SUPER_TENANT_ADMIN},
+        };
+    }
+
+    @BeforeClass(alwaysRun = true, description = "Deploying the Web application")
+    public void init() throws Exception {
+        if (System.getProperty(FrameworkConstants.SYSTEM_PROPERTY_OS_NAME).toLowerCase()
+                .contains(OperatingSystems.WINDOWS.toString().toLowerCase())) {
+            throw new SkipException("Skipping this test case in windows");
+        }
+        super.init(userMode);
+        webAppAdminClient = new WebAppAdminClient(backendURL, sessionCookie);
+        Path webAppFilePath = Paths.get(TestConfigurationProvider.getResourceLocation()
+                , "artifacts", "AS", "security", "manager", "webapp", webAppFileName);
+        webAppAdminClient.uploadWarFile(webAppFilePath.toString());
+
+        assertTrue(WebAppDeploymentUtil.isWebApplicationDeployed(backendURL, sessionCookie, webAppName)
+                , webAppName + " Web Application Deployment failed");
+        webAppUrl = getWebAppURL(WebAppTypes.WEBAPPS) + "/" + webAppName;
+
+        if (userMode == TestUserMode.TENANT_ADMIN || userMode == TestUserMode.TENANT_USER) {
+            if (!webAppUrl.contains("/t/")) {
+                throw new AutomationFrameworkException("Web App Url is not correct for tenants when running test " +
+                        "for tenants " + userInfo.getUserName() + " > " + webAppUrl);
+            }
+        }
+    }
+
+    @Test(groups = "wso2.as", description = "Invoke user store operations through web application")
+    public void testWebAppUserStoreOperations() throws Exception {
+
+        UserManagementClient userMgtClient = new UserManagementClient(backendURL, sessionCookie);
+        userMgtClient.addUser(user1, user1Password, null, defaultProfileName);
+        userMgtClient.addUser(user2, user2Password, null, defaultProfileName);
+
+        userMgtClient.addRole(role1, new String[]{user1, user2}, new String[]{loginPermission}, false);
+        userMgtClient.addRole(role2, new String[]{user1}, new String[]{loginPermission}, false);
+
+        log.info("Users and Roles Added Successfully ");
+        String webAppContext = "/TenantServlet?user=UserStoreTestUser1";
+        String finalURL = webAppUrl + webAppContext;
+
+        URL url = new URL(finalURL);
+        URLConnection connection = url.openConnection();
+        connection.setDoOutput(true);
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+        char[] buffer = new char[Integer.parseInt(connection.getHeaderField("Content-Length"))];
+        int count = buffer.length;
+        String value = null;
+        while ((count = br.read(buffer)) > 0) {
+            value = new String(buffer, 0, count);
+        }
+
+        JSONArray arr = new JSONArray(new String(value));
+
+        ArrayList<String> stringArray = new ArrayList<String>();
```

Comment:
Use appropriate abstract/super types to hide the internals like the exact type of list being used, etc to improve maintainability of the code.